src/Http/Routing/src/EndpointSelectorContext.cs

```diff
@@ -3,68 +3,49 @@
 
 using System;
 using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.Endpoints;
 
 namespace Microsoft.AspNetCore.Routing
 {
-    public sealed class EndpointSelectorContext : IEndpointFeature, IRouteValuesFeature, IRoutingFeature
+    public struct EndpointSelectorContext
```

Comment:
readonly 😄

Reply:
The properties are settable.

Reply:
Hmm, seems to not complain.

Reply:
It's the struct fields that matter. The properties are modifying HttpContext and friends.

Reply:
Yep